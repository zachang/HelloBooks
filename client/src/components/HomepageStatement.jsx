import React from 'react';
import { Link } from 'react-router';

/**
 * HomepageStatement class declaration
 * @class HomepageStatement
 * @extends {React.Component}
 */
export default class HomepageStatement extends React.Component {
  /**
   * Renders Homepage component
   * @return {XML} JSX
   */
  render() {
    return (

      <div className='col s12 statement'>
        <br/><br/>
        <h1 className='header center orange-text'>Welcome!</h1>
        <div className='row center'>
          <h5 className='header col s12 light'>
            Learning is never ending and Knowledge is key.
            So in light of this, we at HelloBooks aim to provide you with all the necessary
            resource needed to expand your mind. Feel free to go through our library.
          </h5>
        </div>

        <div className='row center'>
          <Link to='/register' className='btn-large waves-effect waves-light orange'>
            Join us
          </Link>
          <Link to='/login' className='btn-large waves-effect waves-light orange Login'>
            Login
          </Link>
        </div>
        <br/><br/>
      </div>
    );
  }
}