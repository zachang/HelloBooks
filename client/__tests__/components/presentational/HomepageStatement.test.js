import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import HomepageStatement from '../../../src/components/HomepageStatement.jsx';

const props = {
  btnText: false
};

const wrapper = mount(<HomepageStatement/>);
const shallowWrapper = shallow(<HomepageStatement/>);
const tree = toJson(shallowWrapper);
describe('<HomepageStatement/>', () => {
  it('renders <HomepageStatement/> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it(`should check if the div with 'statement' class exist`, () => {
    expect(wrapper.find('.statement').length).toBe(1);
  });

  it(`should check if 'Join us' link exists`, () => {
    expect(wrapper.find('Link').at(0).text()).toBe('Join us');
  });

  it(`should check if 'Login' link exists`, () => {
    expect(wrapper.find('Link').at(1).text()).toBe('Login');
  });

  it(`should check if the h1 with 'orange-text' class exist`, () => {
    expect(wrapper.find('.orange-text').at(0).text()).toBe('Welcome!');
  });
});
