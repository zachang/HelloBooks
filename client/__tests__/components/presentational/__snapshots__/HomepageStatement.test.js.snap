// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomepageStatement/> renders <HomepageStatement/> component 1`] = `
<div
  className="col s12 statement"
>
  <br />
  <br />
  <h1
    className="header center orange-text"
  >
    Welcome!
  </h1>
  <div
    className="row center"
  >
    <h5
      className="header col s12 light"
    >
      Learning is never ending and Knowledge is key. So in light of this, we at HelloBooks aim to provide you with all the necessary resource needed to expand your mind. Feel free to go through our library.
    </h5>
  </div>
  <div
    className="row center"
  >
    <Link
      className="btn-large waves-effect waves-light orange"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/register"
    >
      Join us
    </Link>
    <Link
      className="btn-large waves-effect waves-light orange"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/login"
    >
      Login
    </Link>
  </div>
  <br />
  <br />
</div>
`;
